### ISSUE TITLE
Go client does not correctly insert, and gives no error

### ISSUE THREAD

**Issue by thomashlvt** [tikv/tikv#3474]
## Bug Report

**What version of Rust are you using?**
1.29.0-nightly

**What operating system and CPU are you using?**
3 servers:
Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz (20-core)
Ubuntu 16.04.5 LTS

client:
Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz (8-core)
Ubuntu 16.04.5 LTS

**What did you do?**
I want to test the performance of TiKV, for this I create a lot of parallel connections and randomly put and get various key-value sizes. For this the RawKVApi Go client is used, and the parallel connections are made by assigning one Go client to each thread.
There are 3 PDs, one on each server, and 3 TiKV nodes, also one on each server (the same servers on which the PDs are running). Everything is set up using Docker (version 17.03.2-ce).
For 400 connections and a key-value size of 256 and 128 bytes respectively there is no problem: keys can be inserted for hours, and when getting them, everything checks out.
However, for the exact same configuration with a key-value size of 256 and 262144 bytes, and only 100 connections (or even lower), keys would not be inserted anymore, without the Go client returning any errors.

**What did you expect to see?**
I expected the Go client to simply correctly put the keys, or when a key was not inserted correctly, to return an error, which could be handled.

**What did you see instead?**
After approximately 7000 keys put, the client logging starts showing these messages:
`time="2018-08-17T08:33:00+02:00" level=info msg="drop regions that on the store 1(<IP>:20160) due to send request fail, err: rpc error: code = Unavailable desc = transport is closing" `
The Go client does not return any errors, but when getting the keys that should be there, they are not (empty values are returned).

Any help would be great, thank you!

**Comment by breeswish** [comment]
Hi, there is currently an issue in our very recent master because one of our new feature is only partially merged and the remaining part is still under review.

Please try run this and restart TiDB:

```bash
pd-ctl -d config set cluster-version 2.1.0
```

@Connor1996 Are there ways to gate our new feature so that users won't fail because of this?

**Comment by Connor1996** [comment]
@breeswish seems no, I will merge it as soon as possible.

**Comment by thomashlvt** [comment]
I tried this.
I also downgraded to a previous release (v2.0.6) of TiDB, but this also didn't help.
With the RawKVClient in go, I put randomly generated keys (10000) of length 256 bytes, and values equal to these keys, appended with zeros up to 262144 bytes. After each Put, I immediately do a Get and check the value, this is always correct, no empty values are returned.
However if I then do a Get for every key (in the same order), the last ~5000 keys (always the exact same amount) consistently return an empty value, even if the Get is sent to only one client, with a single thread. This causes me to believe the values are not present anymore, and there is a data leak.

**Comment by siddontang** [comment]
hi @thomashlvt 

can you show us your test program? do you retry insertion when meeting error?

**Comment by breeswish** [comment]
@thomashlvt Could you try again with our TiKV previous release (2.0.6), or latest (i.e. today's) master? The fix for the issue has been shipped.

**Comment by thomashlvt** [comment]
I found the problem, it was in the client code. My random keys were generated using rand.Intn, but this is also called in the RawKVClient code, making the random numbers not reproducible... (You might consider changing this and using a new rand.Rand struct for generating these numbers so it won't interfere with the code that drives the client. I solved the problem on my side by using this.)

Thank you for your time and the quick responses!
